fsw/src/crypto_props.c

```diff
@@ -0,0 +1,351 @@
+                {   
+                    //reset old values to empty 
+                    memset(p_current->value, 0,sizeof((*pp_self)->VALUE_SIZE_IN_BYTES));
+                    //replace old value with the new 
+                    strcpy(p_current->value,value);  
+                    success = 1; 
+                }
+                else
+                {
+                    //TODO: print error 
+                    printf("ERROR crypto_config_props_set_property_value_helper(),"
+                            "unsafe copy source value size=%lu is larger than destination "
+                            "size=%lu.\n",sizeof(value),sizeof(p_current->value));
+                }
+                break; 
+                
+            }
+            
+        }//end while 
+    }
+    else
+    {
+        printf("ERROR, crypto_config_props_set_property_value_helper() inputs are empty or null.\n");
+    }
+    return success; 
+}
+
+char* crypto_config_get_property_value(char* key)
+{
+    return crypto_config_props_get_property_value_helper(key,&p_self);
+}
+
+char* crypto_config_props_get_property_value_helper(char* key, crypto_config_list** pp_self)
+{
+
+    //iterate through the link list until you reach the right node
+    if (NULL!=key && strlen(key)> 0 && NULL!=(*pp_self))
+    {
+        crypto_config_node* p_current = (*pp_self)->p_head; 
+        while (p_current!=NULL)
+        {
+            //returns 0 if NOT equal 
+            if (strcmp(p_current->key,key)!=0)
+            {
+                p_current= p_current->p_next; 
+            }
+            else
+            {
+                return p_current->value; 
+                
+            }
+            
+        }//end while 
+    }
+    else
+    {
+        printf("ERROR, crypto_config_props_get_property_value_helper() inputs are empty or null.\n");
+    }
+    return "\0"; 
+}
+
+int crypto_config_load_properties(char* file_path)
+{
+    if (NULL!=file_path)
+    {
+        //open file
+        FILE* p_file=NULL;
+        p_file=fopen(file_path,"r");
+        if (NULL!=p_file)
+        {
+            char key[p_self->KEY_SIZE_IN_BYTES];
+            char val[p_self->VALUE_SIZE_IN_BYTES];
+            char current_line[p_self->VALUE_SIZE_IN_BYTES]; 
+            const char* line_delimiter = "="; 
+            int index =-1;
+            //parse the file 
+            while(fgets(current_line, p_self->VALUE_SIZE_IN_BYTES-1,p_file))
+            {
+                // Remove trailing newline
+                current_line[strcspn(current_line, "\n")] = 0;
+                
+                index=strcspn(current_line,line_delimiter);
+                if (index>=0)
+                {
+                    //check for empty commets (lines that start with # or //) and checkf or empty lines
+                    if (current_line[0]!='#' && (current_line[0]!='/' || current_line[1]!='/') && 
+                            strcmp(current_line,"\n") !=0 && strcmp(current_line,"\r\n") !=0 )
```

Comment:
Since this line of code has been the talk of the town... 
- You could run into corner cases where the current_line is of length 1, but you still check the first 2 characters
- You are explicitly checking for "//", so it should be && not || ... If you actually meant to do input validation, then you would do it differently to actually prevent non-alphanumeric characters in the key (EG ', &, $, etc -- / is no different than those in this sense.)